src/libfmt_macros/lib.rs

```diff
@@ -77,9 +77,39 @@ pub struct FormatSpec<'a> {
     pub ty: &'a str,
 }
 
+/// For internal use; same as FormatSpec except using CountInternal instead
+/// of Count. Possible next-references in the count fields will be resolved
+/// before the struct is converted into FormatSpec for returning.
+struct FormatSpecInternal<'a> {
```

Comment:
Could we perhaps not add this and just handle all conversions and such in-place during parsing?